RecoEcal/EgammaCoreTools/interface/EcalClusterLazyTools.h

```diff
@@ -279,8 +281,13 @@ class EcalClusterLazyToolsT : public EcalClusterLazyToolsBase {
   // egamma, but so far covIPhiIPhi hasnt been studied extensively so there
   // could be a bug in the covIPhiIEta or covIPhiIPhi calculations. I dont
   // think there is but as it hasnt been heavily used, there might be one
-  std::vector<float> localCovariances(const reco::BasicCluster &cluster, float w0 = 4.7) const {
-    return ClusterTools::localCovariances(cluster, getEcalRecHitCollection(cluster), topology_, w0);
+  std::vector<float> localCovariances(const reco::BasicCluster &cluster,
```

Comment:
I believe Vincenzo's comment was that here and in other versions of `localCovariances` it might be more efficient to return std::array instead of std::vector, in case the sizes are known at compile time and the vectors are short.

Reply:
okay yes, array will be better in this case, for memory reasons. 
Will it be fine if we do this improvement later, in a separate PR?
This is because, the function `localCovariances` is called from many places, and I need to change all those codes. Moreover, its not only `localCovariances `, but other similar functions can also profit from similar improvement, for example, `covariances `, `scLocalCovariances` etc. So, we wanted to do these technical changes/cleanups in a separate PR. Please let us know if that is okay.

Reply:
For reco it's fine to follow up with this optimization in a later PR, so that this PR is really focused on the new feature, rather than a technical optimization.